src/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs

```diff
@@ -489,7 +489,7 @@ private string[] InternalGetSubKeyNames()
             EnsureNotDisposed();
             int subkeys = InternalSubKeyCount();
             return subkeys > 0 ?
-                InternalGetSubKeyNamesCore(subkeys) :
```

Comment:
Since you have it, you could still pass this through and use it to presize the List<>.